pkg/resource/interfaces.go

```diff
@@ -137,3 +137,15 @@ type Provider interface {
 
 	CredentialsSecretReferencer
 }
+
+// A Target is a Kubernetes object that refers to credentials to connect
+// to a deployment target. Target is a subset of the Claim interface.
+type Target interface {
+	runtime.Object
+	metav1.Object
+
+	LocalConnectionSecretWriterTo
```

Comment:
I believe we will have to either modify `angryjet` or manually implement the necessary methods for `KubernetesTarget` to satisfy this interface due to the fact that the field used is `SecretRef` rather than `WriteConnectionSecretToRef`.

Reply:
I think we'll need to modify angryjet. In reading this I wonder whether we should use a `LocalConnectionSecretReferencer` variant of this interface. We're only specifying where to write the secret in certain modes of operation (propagation from a cluster) - in others we're specifying where to read it from.

Reply:
Could we keep using `LocalConnectionSecretWriterTo` for the existing resources that have a `writeConnectionSecretToRef` field, and add a new `LocalConnectionSecretReferencer` interface that we'd use only for targets with `connectionSecretRef` fields? I imagine the expense would be that we couldn't have `Target` be a strict subset of `Claim`?

Reply:
The primary expense would be that the `Claim` and `Target` interfaces could not use the same managed secret propagation implementation.

Reply:
@negz i think the options we have here are:
1. Implement a separate connection secret propagator for each of Claim and Target (low effort, duplicate code does not feel great though).
1. Use a common interface with methods that access fields with different names (current implementation).
1. Change the name of the Secret field in Target to writeConnectionSecretToRef (this is a weird name if referencing the Secret directly).

Do you have a preference for any of these or an idea for a path I have defined above?

Reply:
Following up on this from today's stand-up: consensus is that for the time being we will use `LocalConnectionSecretWriterTo` in `Target` interface, make the name of the field in the embedded `TargetSpec` struct `WriteConnectionSecretToRef`, but have the JSON tag field name just be `connectionSecretRef`. Will update and add `TODO` to denote that we will likely try to clean this up in the future.